File: src/test/scala/eightyDays/scala/BankSearchTest.scala
package eightyDays.scala

import org.scalatest.WordSpec

class BankSearchTest extends WordSpec {
  "A bank with customers" when {
    "searching for" in {
      val testee = Bank("Test bank")
      val fogg = Person("Fogg", "Phileas")
      val passepartout = Person("Passepartout", "Jean")
      val finnovaLeagEntity = LegalEntity("Finnova", "AG")

      val phileas = testee.addPartner(fogg)
      val jean = testee.addPartner(passepartout)
      val finnova = testee.addPartner(finnovaLeagEntity)

      assert(testee.searchPerson("Jean").nonEmpty)
      assert(testee.searchPerson("Jean").size === 1)
      assert(testee.searchPerson("Jean").head === jean)

      assert(testee.searchPartner("Passepartout").nonEmpty)
      assert(testee.searchPartner("Passepartout").size === 1)
      assert(testee.searchPartner("Passepartout").head === jean)

      assert(testee.searchPerson("Finnova").isEmpty)

      assert(testee.searchPartner("Finnova").nonEmpty)
      assert(testee.searchPartner("Finnova").size === 1)
      assert(testee.searchPartner("Finnova").head === finnova)
    }
  }
}

File: src/test/java/eightyDays/java8/BankSearchTest.java
package eightyDays.java8;

import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class BankSearchTest {
    @Test
    public void getName()  {
        Bank testee = new Bank("Test bank");
        Person fogg = new Person("Fogg", "Phileas");
        Person passepartout = new Person("Passepartout", "Jean");
        LegalEntity finnova = new LegalEntity("Finnova", "AG");

        Identification phileas = testee.addPartner(fogg);
        Identification jean = testee.addPartner(passepartout);
        Identification fin = testee.addPartner(finnova);

        assertFalse(testee.searchPerson("Jean").isEmpty());
        assertFalse(testee.searchPartner("Passepartout").isEmpty());

        assertTrue(testee.searchPerson("Finnova").isEmpty());
        assertFalse(testee.searchPartner("Finnova").isEmpty());
    }
}
